lib/rules/redux-actions-return-objects.js

```diff
@@ -0,0 +1,40 @@
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prohibits returning anything but objects from Redux actions',
+    },
+    fixable: false,
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          whiteList: {
+            default: [],
+            type: 'array',
+          },
+        },
+      },
+    ],
+  },
+
+  create(context) {
+    return {
+      'ReturnStatement'(node) {
+        const filename = context.getFilename();
+        const configuration = context.options[0] || {};
+        const whitelist = configuration.whitelist || [];
+
+        const inActions = filename.includes('state/actions/');
+        if (!inActions) { return; }
+
+        const returnsObject = node.argument && node.argument.type === 'ObjectExpression';
+        if (returnsObject) { return; }
+
+        const whitelisted = !!whitelist.find((w) => filename.includes(w));
```

Comment:
It looks like this code might not perform as expected if given a whitelist that includes `""`.

Reply:
I'm not sure I know what to expect with a whitelist like that. I don't think whitelisting everything in that case is that bad.

Reply:
I think whitelisting everything in that case would be expected, but I think it does the opposite because of `find` combined with `!!`.

Reply:
Ah, good point. I changed it to use `some` instead of `find`. Thanks!